px-context-browser-item.html

```diff
@@ -50,7 +98,120 @@
       canFavorite: {
         type: Boolean,
         value: false
+      },
+      visible: {
+        type: Boolean,
+        value: true
+      }
+    },
+
+    observers: ['_handleFavoritesItemVisible(visible, styleAsFavorite)'],
+
+    _handleFavoritesItemVisible(visible, styleAsFavorite) {
+      if (visible && styleAsFavorite && !this._breadcrumbsEl) {
+        window.requestAnimationFrame(() => {
+          this._breadcrumbsEl = Polymer.dom(this.root).querySelector('#breadcrumbs');
+        });
+      }
+    },
+
+    _computeBreadcrumbsFontStyles(el) {
+      return this.getFontStyles(el);
+    },
+
+    _computeBreadcrumbsWidth(el) {
+      return el.getBoundingClientRect().width;
+    },
+
+    _computeTruncatedBreadcrumbs(breadcrumbs, fontStyles, width) {
+      if (breadcrumbs && fontStyles && width) {
+        return this.truncateBreadcrumbs(breadcrumbs, fontStyles, width);
+      }
+    },
+
+    /**
+     * Reads the computed font styles (font family name and font size) from the targeted
+     * element and returns an object with the details.
+     *
+     * @param {HTMLElement} el
+     * @returns {Object} styles
+     * @returns {String} styles.size - The font size as a string with px suffix (e.g. '12px')
+     * @returns {String} styles.family - The CSS font family declaration as a string (e.g. 'Helvetica')
+     */
+    getFontStyles(el) {
```

Comment:
Should be a private method (e.g. `_getFontStyles`) right?

Reply:
:+1: